src/api/libcellml/generator.h

```diff
@@ -170,92 +66,30 @@ class LIBCELLML_EXPORT Generator: public Logger
-     *
-     * @param model The @c Model to process.
-     */
-    void processModel(const ModelPtr &model);
-
-    /**
-     * @brief Get the @c ModelType of the @c Model.
-     *
-     * Return the @c ModelType of the @c Model processed by this @c Generator.
-     *
-     * @return The @c ModelType.
-     */
-    ModelType modelType() const;
-
-    /**
-     * @brief Get the number of states in the @c Model.
-     *
-     * Return the number of states in the @c Model processed by this
-     * @c Generator.
-     *
-     * @return The number of states.
-     */
-    size_t stateCount() const;
-
-    /**
-     * @brief Get the number of variables in the @c Model.
-     *
-     * Return the number of variables in the @c Model processed by this
-     * @c Generator.
-     *
-     * @return The number of variables.
-     */
-    size_t variableCount() const;
-
-    /**
-     * @brief Get the variable of integration of the @c Model.
-     *
-     * Return the @c Variable of integration of the @c Model processed by this
-     * @c Generator.
-     *
-     * @return The @c Type.
-     */
-    GeneratorVariablePtr voi() const;
-
-    /**
-     * @brief Get the state at @p index.
-     *
-     * Return the state at the index @p index for the @c Model processed by this
-     * @c Generator.
-     *
-     * @param index The index of the state to return.
-     */
-    GeneratorVariablePtr state(size_t index) const;
-
-    /**
-     * @brief Get the variable at @p index.
+     * @brief Get the interface code for the @c AnalyserModel.
      *
-     * Return the variable at the index @p index for the @c Model processed by
-     * this @c Generator.
+     * Return the interface code for the @c AnalyserModel, using its
+     * @c GeneratorProfile.
      *
-     * @param index The index of the variable to return.
-     */
-    GeneratorVariablePtr variable(size_t index) const;
-
-    /**
-     * @brief Get the interface code for the @c Model.
-     *
-     * Return the interface code for the @c Model processed by this
-     * @c Generator, using its @c GeneratorProfile.
+     * @param model The @AnalyserModel for which interface code is to be
+     * generated.
      *
      * @return The code.
      */
-    std::string interfaceCode() const;
+    std::string interfaceCode(const AnalyserModelPtr &model) const;
 
     /**
-     * @brief Get the implementation code for the @c Model.
+     * @brief Get the implementation code for the @c AnalyserModel.
+     *
+     * Return the implementation code for the @c AnalyserModel, using its
+     * @c GeneratorProfile.
      *
-     * Return the implementation code for the @c Model processed by this
-     * @c Generator, using its @c GeneratorProfile.
+     * @param model The @AnalyserModel for which implementation code is to be
+     * generated.
      *
      * @return The code.
      */
-    std::string implementationCode() const;
+    std::string implementationCode(const AnalyserModelPtr &model) const;
```

Comment:
I have had second thoughts on this.  I would like to see the interface:
```
generator->setModel(anaylserModel);
generator->interfaceCode();
generator->implementationCode();
```

Reply:
Hmm, yes, I can see it being more in line with the rest of the API, but that also makes API super annoying to use to me, but ok... 😕

Reply:
Done in commit 0eccc529.